code/game/objects/items/plushies.dm

```diff
@@ -66,6 +67,14 @@
 		playsound(src, hug_sound, 50, 1, -1)
 	if(stuffed || grenade)
 		src.visible_message("<span class='notice'>\The [src] gives \the [M] a [pick("hug", "warm embrace")].</span>")
+		if(hug_reagents && M.reagents)
```

Comment:
the structure of this isn't good

first, use `hug_reagents.len` instead because an empty list still exists
second, why is this even a split tree? shouldn't it check to see if the target has the reagent regardless of whether it's the user self-clicking or someone else?

Reply:
> first, use `hug_reagents.len` instead because an empty list still exists
Got it
> second, why is this even a split tree? shouldn't it check to see if the target has the reagent regardless of whether it's the user self-clicking or someone else?
Didn't get it

Reply:
you're branching the if statement one line below this to see if the recipient of the hug is the one using it, then only in the case of the user hugging himself is there a check to see if he has that reagent already

Reply:
i, a completely unrelated person, believe the split is there to differentiate between hugging yourself and hugging someone else with the plushie
hugging someone else with your hands doesn't limit the paracetamol and neither do the plushies
while hugging yourself is impossible plushyless(unless you're a monkey), while plushies have a 1u maximum

Reply:
this has DD in it